tools/create_minimal_python.py

```diff
@@ -0,0 +1,73 @@
+from importlib import machinery
+import os
+import shutil
+import sys
+
+# These are modules which otherwise *should* be removed, but can't since they
+# are requirements of pytest.
+
+# TODO: Find a solution to this, possibly involving running tests in a
+# separate, more limited process.
+
+PYTEST_REQUIREMENTS = [
+    '_posixsubprocess', 'select', 'binascii', 'pyexpat', '_socket'
+]
+
+# These are optional parts of the standard library that are required for core
+# functionality in matplotlib that can not easily be avoided.
+
+MATPLOTLIB_HARD_REQUIREMENTS = [
+    'zlib', '_csv'
+]
+
+
+KEEP_MODULES = PYTEST_REQUIREMENTS + MATPLOTLIB_HARD_REQUIREMENTS
+
+
+def iterate_modules_to_remove():
+    with open(os.path.join(
+            os.path.dirname(__file__), 'Setup.dist'), 'r') as fd:
+        for line in fd:
+            if line.startswith('# Modules with some UNIX dependencies'):
+                break
+
+        for line in fd:
+            line = line.strip()
+            if line.startswith('#') and not line.startswith('# ') and len(line) > 1:
+                modulename = line[1:].split()[0]
+                if modulename not in KEEP_MODULES:
+                    yield modulename
+
+    # Some other commonly-missing modules that aren't in Setup.dist
```

Comment:
If you're going to hardcode a list there you may as well do without Setup.dist and just hardcode the entire list...

Reply:
Perhaps.  Some of the stuff is optional due to Setup.dist, and some things are turned on/off by environment tests in CPython's configure script.  The latter is harder to pick out automatically, but maybe not impossible.